### DISCUSSION TITLE
VPN Protocols

### DISCUSSION THREAD

**Discussion by majeeddl** [mudler/edgevpn#53] [General]
What are protocols for VPN?

**Comment by mudler** [comment]
Hi! :wave: Can you be a bit more specific? EdgeVPN uses libp2p, so uses all the transport enabled by default in libp2p. This means typically means UDP/TCP and UDP/quic might be used behind the scene

**Reply by fischerdouglas** [reply]
Maybe my doubt is the same as my colleague's, and we are having difficulty understanding and expressing our doubt.

But comparing EdgeVPN with Tail/Scale, ZeroTier, or even OpenVPN, the network overlay layer becomes explicit.

That is, in these other tools we have network addresses, routes, and depending on the case we can even involve routing protocol in certain nodes.

If I understood so far, and I am still reading documentation (and sometimes some part of the code, but I am not good at it), the purpose of EdgeVPN is not to deal with this tunneling and routing.

It simply maps ports like in SSH tunneling, or like in ngrok (I think I saw it mentioned in the documentation), and that's it.

Is that right? Did I understand correctly?
